### DISCUSSION TITLE
Whois query problem on warning received

### DISCUSSION THREAD

**Discussion by tradexsrl** [netdata/netdata#17994] [Q&A]
Hi I have setup Whois query on a bunch of domain, the expiration is end of July 2024 , mid September 2024 and mid 2025
I received 3 warning for 3 domain that will expire in 85 days or 40 days, but I don't understand why because default is 30 days and 15 days 
Another question is why the expiration is in seconds? I get for example expiration in 7468788 which is not very helpful, so i need to convert it into days

**Comment by ilyam8** [comment]
@tradexsrl hey

> but I don't understand why because default is 30 days and 15 days

The default is 90 (warning) and 30 (critical). I don't remember why we decided to use these numbers, but they were always 90/30, we haven't changed them since we added _whoisquery_.

> Another question is why the expiration is in seconds?

That is expected. The metric (_whoisquery.time_until_expiration_) is in seconds. Displaying the value (seconds) in "days:hours:minutes" format is a UI feature.

**Comment by tradexsrl** [comment]
if you look in at this url: https://learn.netdata.cloud/docs/collecting-metrics/synthetic-checks/domain-expiration-date?_gl=1*1tsonez*_gcl_au*MTY1MjczNjY4OS4xNzEyMjIzNjMy*_ga*NzQ3NjA4MzU4LjE3MTIyMjM2MzI.*_ga_J69Z2JCTFB*MTcxOTIxODMyMS4zNS4wLjE3MTkyMTgzMjMuNTguMC4w

it says 30 days and 15 days

to display in days:hours:minutes in ui how can be done ?

**Comment by ilyam8** [comment]
I changed seconds to days in #17999.

<img width="1294" alt="Screenshot 2024-06-24 at 11 50 02" src="https://github.com/netdata/netdata/assets/22274335/81c55054-2226-4942-85d2-e9db6118243c">

---

> if you look in at this URL:

Wrong documentation. I will update it.

When using WhoisQuery, is 90 days a good timeframe for a notification, or would a shorter window, like 30 days, be better?

**Reply by ilyam8** [reply]
Changed 90/30 to 30/15 in #18000.

**Reply by tradexsrl** [reply]
in my opinion 30/15 is better

**Reply by tradexsrl** [reply]
after update to 1.4.6 i have no more whois query section.
i have attached a log but i haven't seen any error that broke it:
[log.txt](https://github.com/user-attachments/files/15955800/log.txt)

**Reply by tradexsrl** [reply]
> I changed seconds to days in #17999.
> <img alt="Screenshot 2024-06-24 at 11 50 02" width="1294" src="https://private-user-images.githubusercontent.com/22274335/342268449-81c55054-2226-4942-85d2-e9db6118243c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkyNDQzMTcsIm5iZiI6MTcxOTI0NDAxNywicGF0aCI6Ii8yMjI3NDMzNS8zNDIyNjg0NDktODFjNTUwNTQtMjIyNi00OTQyLTg1ZDItZTlkYjYxMTgyNDNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI0VDE1NDY1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3M2ZiNDNjNjNjMTBkNWQxZTAwZWQ4NjQyNDBkZjY2YjMzYmM5OGMzMzEyYzBjZWM3ZWQ4YWU4NTExNzY5OGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TOuRHe-pcpu0Ed7KfFd9BL1M-lTasG9XmPpoQwRAznU">
> 
> > if you look in at this URL:
> 
> Wrong documentation. I will update it.
> 
> When using WhoisQuery, is 90 days a good timeframe for a notification, or would a shorter window, like 30 days, be better?

I continue to see in seconds:
![immagine](https://github.com/netdata/netdata/assets/62424179/d9694f58-732b-4573-b5fa-c6531d9534c8)

**Reply by ilyam8** [reply]
@tradexsrl that is expected. The changes I made are in the master branch.

- They will be in the nightly release tomorrow.
- They will be in the stable version when we release the next stable version (I don't know when).

If you are using stable and want to get the changes, you need to do:

1. Modify `/usr/lib/netdata/conf.d/health.d/whoisquery.conf` ([changes](https://github.com/netdata/netdata/pull/17999/files#diff-52ea26bf1d2243fad56749f8020a7625c6a2f0dd7a51eef4aed19361cb5452f2)).
2. Update all your WhoisQuery jobs like

  ```yaml
  jobs:
    - name: example
      source: example.org
      days_until_expiration_warning: 30 # add this
      days_until_expiration_critical: 15 # and this
  ```
3. Restart Netdata

**Reply by tradexsrl** [reply]
and what aboout the visualization in seconds and not in days as shown in the screenshot?

**Reply by ilyam8** [reply]
1st point in my previous answer.

**Reply by tradexsrl** [reply]
Is it not possible to modify the chart on netdata.cloud as i will do with for example grafana?

**Reply by ilyam8** [reply]
I don't understand your question.

**Reply by tradexsrl** [reply]
without waiting stable relase , can i modify the chart relative to the whoisquuery to show days and not seconds? or modify the alert section to show days and not seconds also when received mail message?

**Reply by ilyam8** [reply]
The chart shows days and not seconds. The alert shows seconds. I answered your question [here](https://github.com/netdata/netdata/discussions/17994#discussioncomment-9862394)
